.github/GUIDE.md

```diff
@@ -0,0 +1,108 @@
+
+## Architecture
+
+```bash
+⎪ actions
+├── file-diff
+├──── action.yml - this defines the inputs and outputs of the action
+├──── index.js - the code that runs the action
+├──── package.json - has dependencies so has package
+├──── README.md - more documentation yay!
+⎪ workflows
+├── development.yml - run on pull requests only
+├── production.yml - runs only on pushes to main
+├── vrt.yml - a reusable workflow that can be called by other workflows (i.e., development.yml or production.yml) or called on it's own via [workflow dispatch](https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/).
+├── build.yml - builds a branch and outputs the compiled assets as artifacts
+├── compare-results.yml - compares the compiled assets between the pull request branch and the main branch
+```
+
+**But wait! There's more!**
+
+```bash
+⎪ ISSUE_TEMPLATE
+├── --bug-report.md
+├── --documentation-issue.md
+├── --feature-request.md
+├── --support-request.md
+⎪ PULL_REQUEST_TEMPLATE.md
+⎪ CONTRIBUTING.md
+⎪ dependabot.yml
+```
+
+## Actions
+
+### File Diff
+
+This action is used to determine if a compiled asset has changed between two branches. See the [README](./file-diff/README.md) for more information.
+
+## Workflows
+
+### Development
+
+This workflow runs:
+
+- on pull requests when:
+  - opened against the `main` branch
+  - opened, reopened, synchronized (i.e., when a commit is pushed to the pull request), labeled or unlabeled, or if auto merge is enabled
+  - any files other than markdown have changed (i.e., will not run on a pull request that only changes markdown files)
+
+#### What does this workflow do?
+
+##### 👷‍♀️ Build
+
+Builds the pull request branch against various development environments. Installs dependencies and builds the project looking for basic processing errors.
+
+##### 👷‍♀️ Compare results
+
+Compares the compiled assets between the pull request branch and the base branch. If there are differences, a comment is added to the pull request with a table detailing the files and the size differences.
+
+_to-do_: This needs to diff the actual content of the files as well. Right now we're leveraging a canary approach which would catch any file size changes to the compiled assets. However, **if the content of the file changes but the size doesn't, we won't catch that**.
+
+##### 🧹 Linting
+
+Runs stylelint or eslint if any relevant assets have been updated in this PR.
+
+##### 📝 Publish site
+
+After the build and visual regression tests have passed, this will build the docs site and publish it to Netlify.
+
+##### 📸 Visual regression testing
+
+Run these tests if the `run_vrt` label is added to the pull request.
+
+**OR** the pull request is not in a draft state and is mergeable (meaning no conflicts with the base branch)
+
+**OR** the pull request review request is approved.
```

Comment:
So this means VRTs will run automatically if a PR is open and mergeable, you don't have to add the `run_vrt` label? Just want to double check that this is the desired behavior. Doesn't look like the code for this in the workflow has changed, so this must be how it's currently set up?

Reply:
It's how it's supposed to work yes - honestly it's not working that way and is something on my to-do list to look into 🙃